### PR TITLE
feat: Use client.saveAll to save transaction categories

### PR DESCRIPTION
**Pull request by ptbrowne**
Depends on https://github.com/cozy/cozy-client/pull/995

Now that cozy client has saveAll, we use it instead of calling
repeatedly save.

## Failure

Batch edits can fail because of conflicts (a version of a document has changed without cozy-client knowing about it) or document being deleted.

In case of failure, operations being recategorized are refetched to mitigate the chance of a future conflict.

Here is a video where I try to recategorize 3 operations and I force a conflict

- One of the operations is changed "behind the back" of cozy-client, so the version that cozy-client has is stale
- We can see that the first batch edit fails somehow (but two operations have been recategorized correctly)
- When the failure happens, operations are refetched and we see that cozy-client now has the correct version for the 3 operations
- The subsequent batch edit works since cozy-client has the latest version.

https://user-images.githubusercontent.com/465582/128198486-6b1b2eca-b00f-42c4-9965-24d534c7d16b.mp4

### CODE REVIEW

File: src/ducks/transactions/TransactionModal/TransactionCategoryEditorDialog.jsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ const TransactionCategoryEditorDialog = ({ transaction, onClose }) => {
   return (
     <TransactionCategoryEditor
       beforeUpdates={handlePop}
-      afterUpdate={onAfterUpdate}
+      afterUpdates={onAfterUpdate}
       onCancel={handlePop}
       transactions={[transaction]}
     />
```
Comment (JF-Cozy) [inline src/ducks/transactions/TransactionModal/TransactionCategoryEditorDialog.jsx:32]: nit: On peut peut-être renommer `onAfterUpdate` en  `onAfterUpdates` ?

Comment (ptbrowne) [reply src/ducks/transactions/TransactionModal/TransactionCategoryEditorDialog.jsx:32]: Done

File: src/ducks/transactions/TransactionCategoryEditor.jsx [hunk 2]
```diff
@@ -1,58 +1,59 @@
...
 
-      await Promise.all(promises)
+      try {
+        await client.saveAll(newTransactions)
+      } catch (e) {
+        const qdef = Q(TRANSACTION_DOCTYPE).getByIds(
+          newTransactions.map(t => t._id)
+        )
+
+	// eslint-disable-next-line no-console
```
Comment (flohhhh) [inline src/ducks/transactions/TransactionCategoryEditor.jsx:42]: nit: Indent comment

Comment (ptbrowne) [reply src/ducks/transactions/TransactionCategoryEditor.jsx:42]: Done.

**Review by JF-Cozy — APPROVED** [review]

**Pull request merged**